store/tikv/region_cache.go

```diff
@@ -385,6 +385,9 @@ func (c *RegionCache) loadStoreAddr(bo *Backoffer, id uint64) (string, error) {
 	for {
 		store, err := c.pdClient.GetStore(bo.ctx, id)
 		if err != nil {
+			if errors.Cause(err) == goctx.Canceled {
+				return "", errors.Trace(err)
+			}
```

Comment:
I change here just because unit test assert the returned error to be cancelled,
if those lines are not added, the returned error will be changed to timeout by backoff, 
unit test would fail then.

This PR should just do one thing, which is "context cancel should not trigger onSendFail and drop cache" as the title described. Real fix of "why context is cancelled" will be in the following PR, so I prefer minimal changes here.